packages/fxa-content-server/app/scripts/lib/crypto/totp.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import base32Decode from 'base32-decode';
+
+function trimOrPad(num, digits) {
+  const str = num.toString().substr(-digits);
+  if (str.length === digits) {
+    return str;
+  }
+  return new Array(digits - str.length + 1).join('0') + str;
+}
+
+export function getCode(secret, digits = 6, now = Date.now()) {
```

Comment:
This kind of makes me wonder if we should replace [otplib](https://github.com/yeojz/otplib) in auth-server with this implementation. 

Either way, this is 💯